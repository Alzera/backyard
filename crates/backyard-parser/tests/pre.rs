use backyard_parser::parse;

#[test]
fn basic() {
  let asts = parse(true, "++$a;").unwrap();
  insta::assert_yaml_snapshot!(asts);
}

#[test]
fn negate() {
  let asts = parse(true, "!$a;").unwrap();
  insta::assert_yaml_snapshot!(asts);
}

#[test]
fn silent() {
  let asts = parse(true, "@$a();").unwrap();
  insta::assert_yaml_snapshot!(asts);
}

#[test]
fn ellipsis() {
  let asts = parse(true, "...$a();").unwrap();
  insta::assert_yaml_snapshot!(asts);
}

#[test]
fn ambersand() {
  let asts = parse(true, "$a = &$b;").unwrap();
  insta::assert_yaml_snapshot!(asts);
}
