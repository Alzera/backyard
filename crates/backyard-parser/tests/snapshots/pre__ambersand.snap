---
source: crates/backyard-parser/tests/pre.rs
expression: asts
snapshot_kind: text
---
node_type: program
children:
  - node_type: assignment
    left:
      node_type: variable
      name:
        node_type: identifier
        name:
          - 97
        loc:
          start:
            line: 1
            column: 0
            offset: 0
          end:
            line: 1
            column: 1
            offset: 1
        leadings: ~
        trailings: ~
      loc:
        start:
          line: 1
          column: 0
          offset: 0
        end:
          line: 1
          column: 1
          offset: 1
      leadings: ~
      trailings: ~
    operator: Default
    right:
      node_type: reference
      statement:
        node_type: variable
        name:
          node_type: identifier
          name:
            - 98
          loc:
            start:
              line: 1
              column: 6
              offset: 6
            end:
              line: 1
              column: 7
              offset: 7
          leadings: ~
          trailings: ~
        loc:
          start:
            line: 1
            column: 6
            offset: 6
          end:
            line: 1
            column: 7
            offset: 7
        leadings: ~
        trailings: ~
      loc:
        start:
          line: 1
          column: 5
          offset: 5
        end:
          line: 1
          column: 6
          offset: 6
      leadings: ~
      trailings: ~
    loc:
      start:
        line: 1
        column: 3
        offset: 3
      end:
        line: 1
        column: 6
        offset: 6
    leadings: ~
    trailings: ~
loc:
  start:
    line: 1
    column: 0
    offset: 0
  end:
    line: 1
    column: 8
    offset: 8
leadings: ~
trailings: ~
